### DISCUSSION TITLE
Reading the dye temperature

### DISCUSSION THREAD

**Discussion by goldfingyr** [jrowberg/i2cdevlib#737] [Q&A]
So far I can see that method hasn't been implementet (dmpGetTemperature).
Correct?
And if so, are there any plans for inclusion?

Cheers,
Karsten

**Comment by ZHomeSlice** [comment]
embedded DMP firmware doesn't have the temperature value available. The eDMP firmware takes over the FIFO buffer and loads its own data into it.  if you choose not to use the eDMP firmware you could load the temperature value along with accel and gyro data into the FIFO buffer but the Quaternions and filtered accel and gyro will not be available.
you could just get the temperature directly 

/** Get current internal temperature.
 * @return Temperature reading in 16-bit 2's complement format
 * @see MPU9250_RA_TEMP_OUT_H
 */
int16_t MPU9250::getTemperature() {
    I2Cdev::readBytes(devAddr, MPU9250_RA_TEMP_OUT_H, 2, buffer);
    return (((int16_t)buffer[0]) << 8) | buffer[1];
}
Z
